Handle langword for see elements

martincostello opened the pull request:
# Handle `langword` for `see` elements

Handle `<see langword="value" />` references in XML documentation as inline code.

## Description

Extract the value from `langword` on `see` references and emit as inline code.

For example `<see langword="true" />` becomes ````true````.

Questions:

1. Should it also handle `seealso`?
2. ~~Do we output verbatim, or map the keywords to something? (e.g. C# `null` to `null` but VB `Nothing` to `null`)~~

Fixes #61042.

martincostello commented on src/OpenApi/gen/XmlComments/XmlComment.cs:
```diff
@@ -197,6 +202,35 @@ private static void ResolveCrefLink(Compilation compilation, XNode node, string
+                continue;
+            }
+
+            // TODO Is this the right thing to do? For example C# would get `true`, `false`, `null`
```
@captainsafia any thoughts on this so I can get this PR over the line?

captainsafia replied:
WRT to normalization, I think the [C# keywords](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/) are probably sufficient to include in the document as-is.

WRT to VB in particular, I'm inclined to say we should consider limiting this generator to just C# (`[Generator(LanguageNames.CSharp)]`) since I haven't done any validation on VB-based projects and it would be a good safety-guard for anywhere we are doing CSharp specific syntax checks.

In practice, I think the keywords most likely to appear in API comments are probably things like true/false/null.

martincostello replied:
Have opened #62069 with the proposed change to disable anything but C#.

martincostello committed:
Handle langword for see elements

Handle `<see langword="" />` references in XML documentation as inline code.

Resolves #61042.

martincostello committed:
Remove redundant using

Added back by merge/rebase but not needed.

captainsafia commented on src/OpenApi/test/Microsoft.AspNetCore.OpenApi.SourceGenerators.Tests/CompletenessTests.cs:
```diff
@@ -437,6 +437,39 @@ public static T GetGenericValue<T>(T para)
+    /// </summary>
+    /// <param name="disposing">
+    /// <see langword="true" /> to release both managed and unmanaged resources;
+    /// <see langword="false" /> to release only unmanaged resources.
```
```suggestion
    /// <see langword="false" /> to release only unmanaged resources.
    /// <see langword="null" /> to indicate a no-op.
```

captainsafia commented on src/OpenApi/gen/XmlComments/XmlComment.cs:
```diff
@@ -197,6 +202,33 @@ private static void ResolveCrefLink(Compilation compilation, XNode node, string
     }
 
+    /// <summary>
+    /// Resolves the links in the XML documentation into type names.
```
```suggestion
    /// Resolves the links in the XML documentation into language keywords.
```

captainsafia commented on src/OpenApi/gen/XmlComments/XmlComment.cs:
```diff
@@ -197,6 +202,33 @@ private static void ResolveCrefLink(Compilation compilation, XNode node, string
+    /// Resolves the links in the XML documentation into type names.
+    /// </summary>
+    /// <param name="node">The target node to process crefs in.</param>
+    /// <param name="elementName">The node type to process crefs for, can be `see` or `seealso`.</param>
```
```suggestion
    /// <param name="elementName">The node type to process langwords for, can be `see` or `seealso`.</param>
```

captainsafia reviewed (approved):
Looks good overall! A few small nits to polish this up.

martincostello committed:
Apply suggestions from code review

Co-authored-by: Safia Abdalla <safia@safia.rocks>
